app/adapters/alma_adapter.rb

```diff
@@ -36,10 +36,22 @@ def get_availability(ids:)
   end
 
   # Returns availability for one bib id
-  def get_availability_one(id:)
+  def get_availability_one(id:, deep_check: false)
     bibs = Alma::Bib.find(Array.wrap(id), expand: ["p_avail", "e_avail", "d_avail", "requests"].join(",")).each
     return nil if bibs.count == 0
-    { bibs.first.id => AvailabilityStatus.new(bib: bibs.first).bib_availability }
+
+    av_info = AvailabilityStatus.new(bib: bibs.first).bib_availability
+
+    temp_locations = av_info.any? { |_key, value| value[:temp_location] }
+    if temp_locations && deep_check
```

Comment:
This is the new behavior that the client can request via the deep_check parameter to drill down for more information if the initial call reports holdings in temporary locations.